src/App.tsx

```diff
@@ -15,49 +17,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  isReversed: boolean,
+  sortType: SortType;
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType === SortType.ALPHABET) {
+    visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
+  }
+
+  if (sortType === SortType.LENGTH) {
+    visibleGoods.sort((good1, good2) => good1.length - good2.length);
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+export const App = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getReorderedGoods(
+    goodsFromServer,
+    { isReversed, sortType },
+  );
+
   return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+    <>
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-info',
+              { 'is-light': sortType !== SortType.ALPHABET },
+            )}
+            onClick={() => setSortType(SortType.ALPHABET)}
```

Comment:
it is bad practise to make anonymous arrow function in jsx
consider to make setSort handler which will accept sortType and return arrow function and call here and for sort by length button.
```
  const setSort = (sortType: SortType) => () => setSortType(sortType);
```
```suggestion
            onClick={setSort(SortType.ALPHABET)}
```